.github/workflows/windows-macos.yml

```diff
@@ -166,6 +169,48 @@ jobs:
       with:
         vcpkgDirectory: '${{ github.workspace }}/3rd-party/vcpkg'
 
+      # TODO: remove this when self-hosted runners are dropped
+    - name: Disable vcpkg binary cache on self-hosted runners
+      id: disable-cache
+      shell: bash
+      run: |
+        if [[ "${{ matrix['runs-on'] }}" == *multipass* ]]; then
+          echo "VCPKG_BINARY_SOURCES=" >> $GITHUB_ENV
+        fi
+
+    - name: Install nuget
+      if: ${{ runner.os == 'macOS' && matrix.arch == 'arm64' && env.VCPKG_BINARY_SOURCES }}
+      run: |
+        brew install nuget mono
+
+    - name: Configure NuGet Source
+      if: ${{ env.VCPKG_BINARY_SOURCES }} # TODO: remove this when self-hosted runners are dropped
+      id: setup_nuget
+      continue-on-error: true
+      shell: 'bash'
+      run: |
+        ${{ matrix.mono }} "$($VCPKG_ROOT/vcpkg fetch nuget | tail -n 1)" \
```

Comment:
In order to not lose context we can put a link to the official guide as comment:

https://learn.microsoft.com/en-us/vcpkg/consume/binary-caching-github-packages?pivots=linux-runner

Reply:
Done